rectools/models/sasrec.py

```diff
@@ -52,6 +51,14 @@ def forward(self, seqs: torch.Tensor, timeline_mask: torch.Tensor, attn_mask: to
         raise NotImplementedError()
 
 
+class PositionalEncodingBase(torch.nn.Module):
+    """Base class for positional encoding. Used only for type hinting."""
```

Comment:
If any class is used only for type hinting let's use instead [protocols](https://docs.python.org/3/library/typing.html#typing.Protocol) - it's less confusing and you don't need to implement inheritance and to write such comments

Reply:
@blondered FYI

Reply:
next pr